ddht/v5_1/alexandria/partials/chunking.py

```diff
@@ -0,0 +1,224 @@
+import itertools
+from typing import Iterable, List, Sequence, Tuple
+
+from eth_typing import Hash32
+from eth_utils import to_tuple
+from eth_utils.toolz import sliding_window
+from ssz.constants import CHUNK_SIZE, ZERO_BYTES32
+
+from ddht.v5_1.alexandria.constants import GB, POWERS_OF_TWO
+from ddht.v5_1.alexandria.partials.typing import TreePath
+
+
+@to_tuple
+def compute_chunks(data: bytes) -> Iterable[Hash32]:
+    """
+    An optimized version of SSZ chunking specifically for byte
+    strings.
+
+    Takes the data and splits it into chunks of length 32.  The last chunk is
+    padded out to 32 bytes with null bytes `0x00` if it is not full.
+    """
+    if not data:
+        yield ZERO_BYTES32
+        return
+    elif len(data) > GB:
+        raise Exception("too big")
+    data_length = len(data)
+    if data_length % CHUNK_SIZE == 0:
+        padded_data = data
+    else:
+        padding_byte_count = CHUNK_SIZE - data_length % CHUNK_SIZE
+        padded_data = data + b"\x00" * padding_byte_count
+
+    padded_length = len(padded_data)
+    for left_boundary, right_boundary in sliding_window(
+        2, range(0, padded_length + 1, CHUNK_SIZE)
+    ):
+        yield Hash32(padded_data[left_boundary:right_boundary])
```

Comment:
Again, no bug here, but this section could become a little shorter (and likely much more inefficient):

```python
    from eth_utils.toolz import partition
    ...
    for chunk in partition(CHUNK_SIZE, padded_data):
        yield Hash32(bytes(chunk))
```

Reply:
Slicing up byte strings should be easier... I've wondered about using `memoryview` to reduce copies.  

For now I'm doing my best to avoid going down performance rabbit holes until we can profile the full client in a production setting and understand where our biggest gains can be had.